src/main/default/classes/TriggerHandler.cls

```diff
@@ -65,51 +73,66 @@ public class TriggerHandler {
-                disabledHandlers.add(handlerName);
-            } else if (handlerName != null && objectName != null) {
-                if (!disabledObjectHandlers.containsKey(objectName)) {
-                    disabledObjectHandlers.put(objectName, new Set<String>());
-                }
-                disabledObjectHandlers.get(objectName).add(handlerName);
+    private void doAutomaticBinding() {
+        if (!triggerConfig.enabled) {
+            return;
+        }
+        for (String handlerName : triggerConfig.handlerOrder) {
+            TriggerConfigService.HandlerConfig handlerConfig = triggerConfig.handlers.get(handlerName);
+            // do not bind if auto bind is off or if handler is disabled
+            if (!handlerConfig.autoBind || !handlerConfig.enabled) {
+                continue;
+            }
+            Type t = getType(handlerConfig.className);
+            if (t == null) {
+                System.debug(handlerConfig.className + '? Looks like that handler class does not exist!');
+                continue;
+            }
+            HandlerInterface handlerClass = (HandlerInterface) t.newInstance();
+            if (handlerConfig.beforeInsert) {
+                bind(Evt.beforeInsert, handlerClass);
+            }
+            if (handlerConfig.afterInsert) {
+                bind(Evt.afterInsert, handlerClass);
+            }
+            if (handlerConfig.beforeUpdate) {
+                bind(Evt.beforeUpdate, handlerClass);
+            }
+            if (handlerConfig.afterUpdate) {
+                bind(Evt.afterUpdate, handlerClass);
+            }
+            if (handlerConfig.beforeDelete) {
+                bind(Evt.beforeDelete, handlerClass);
+            }
+            if (handlerConfig.afterDelete) {
+                bind(Evt.afterDelete, handlerClass);
+            }
+            if (handlerConfig.afterUndelete) {
+                bind(Evt.afterUndelete, handlerClass);
             }
         }
     }
-    static {
-        cacheTriggerControl([SELECT Class__c, Object__c FROM Trigger_Control__mdt]);
-    }
 
-    private Boolean objectDisabled(String xSObjType) {
-        return disabledObjects.contains(xSObjType);
+    private Type getType(String className) {
+        Type t;
+        try {
+            t = Type.forName(className); // doesn't throw exception if invalid name!
+        } catch (Exception e) {
+            // not sure we need a try/catch
+            System.debug(e.getMessage());
+        }
+        return t;
     }
 
     private Boolean handlerDisabled(HandlerInterface h) {
-        return disabledHandlers.contains(getHandlerName(h));
-    }
-
-    private Boolean objectHandlerDisabled(String xSObjectType, HandlerInterface h) {
-        return disabledObjectHandlers.containsKey(xSObjectType) &&
-            disabledObjectHandlers.get(xSObjectType).contains(getHandlerName(h));
+        return triggerConfig.disabledHandlers.contains(getHandlerName(h));
```

Comment:
i think this might come up with the old style of handler disablement without a config object, i honestly think we can drop it completely though, i don't think people will really disable specific directly bound handlers, they'll disable a specific event or an entire object

Reply:
Really?

I think it seems more likely that you would disable a trigger handler than anything else...

Turning the handler off by unchecking events means you need to remember which events to "rebind" it to when you go back to turn it on.

Reply:
you're confusing turning off a handler for an object with the original version where you could turn off a handler globally for all objects, i'm referring to the later not being necessary. what you're describing is absolutely necessary. i'm trying to agree with you on getting rid of `TriggerConfig.disabledHandlers`

Reply:
I like it ➕

Reply:
Disabled handlers gone